bndtools.builder/src/org/bndtools/builder/classpath/BndContainerInitializer.java

```diff
@@ -255,26 +273,59 @@ public static void setClasspathEntries(IJavaProject javaProject, IClasspathEntry
                     if (c.getType() == Container.TYPE.PROJECT) {
                         IResource resource = ResourcesPlugin.getWorkspace().getRoot().getFile(p);
-                        List<IAccessRule> rules = projectAccessRules.get(c.getProject());
-                        IAccessRule[] accessRules = null;
-                        if (rules != null) {
-                            rules.add(JavaCore.newAccessRule(new Path("**"), IAccessRule.K_NON_ACCESSIBLE));
-                            accessRules = rules.toArray(new IAccessRule[rules.size()]);
+                        List<IAccessRule> rules = projectAccessRulesExports.get(c.getProject());
+                        if (rules != null && rules.size() > 0) {
+                            IAccessRule[] accessRules = rules.toArray(new IAccessRule[rules.size()]);
+                            cpe = JavaCore.newProjectEntry(resource.getProject().getFullPath(), accessRules, false, extraAttrs, true);
+                            result.add(cpe);
+                        }
+                        rules = projectAccessRulesPrivates.get(c.getProject());
+                        if (rules != null && rules.size() > 0) {
+                            IAccessRule[] accessRules = rules.toArray(new IAccessRule[rules.size()]);
+                            cpe = JavaCore.newProjectEntry(resource.getProject().getFullPath(), accessRules, false, extraAttrs, false);
+                            result.add(cpe);
                         }
-                        cpe = JavaCore.newProjectEntry(resource.getProject().getFullPath(), accessRules, false, extraAttrs, true);
                     } else {
-                        IAccessRule[] accessRules = calculateRepoBundleAccessRules(c);
-                        cpe = JavaCore.newLibraryEntry(p, null, null, accessRules, extraAttrs, false);
+                        calculateRepoBundleAccessRules(c, model, exports);
+                        if (exports.size() > 0) {
+                            IAccessRule[] accessRules = exports.toArray(new IAccessRule[exports.size()]);
+                            cpe = JavaCore.newLibraryEntry(p, null, null, accessRules, extraAttrs, true);
+                            result.add(cpe);
+                        }
+                        cpe = JavaCore.newLibraryEntry(p, null, null, null, extraAttrs, false);
+                        result.add(cpe);
                     }
-                    result.add(cpe);
                 }
             } else {
                 errors.add(c.getError());
             }
         }
 
+        //[cs] set project variable: "accessrulesdebug: true" to enable some extra eclipse output for debugging.
+        if (Boolean.parseBoolean(model.getProperty("accessrulesdebug", "false"))) {
+            System.out.println("IClasspathEntrys: project = " + model.getName());
+            for (IClasspathEntry f : result) {
+                System.out.println("--- " + f);
+            }
+        }
+
```

Comment:
I thought it was useful for debugging purposes. I keep adding code like that to aid debugging. Might as well add it to the baseline.